--- ../usr/lib/setup/setup	2024-05-17 15:17:07.207384436 +0300
+++ setup	2024-05-17 15:35:28.683418050 +0300
@@ -50,15 +50,15 @@
 # Before probing, activate any LVM partitions
 # that may exist from before the boot:
 vgchange -ay 1> /dev/null 2> /dev/null
-if probe -l 2> /dev/null | grep -E 'Linux$' 1> /dev/null 2> /dev/null ; then
- probe -l 2> /dev/null | grep -E 'Linux$' | sort | uniq 1> $TMP/SeTplist 2> /dev/null
+if probe -l 2> /dev/null | grep -E 'Linux$|Bcachefs$' 1> /dev/null 2> /dev/null ; then
+    probe -l 2> /dev/null | grep -E 'Linux$|Bcachefs$' | sort | uniq 1> $TMP/SeTplist 2> /dev/null
 else
- dialog --title "NO LINUX PARTITIONS DETECTED" \
- --msgbox "There don't seem to be any partitions on this machine of type \
-Linux. You'll need to make at least one of these to install Linux. \
-To do this, you'll need to leave 'setup', and make the partitions using \
-'cfdisk' (MBR partitions) or 'cgdisk' (GPT partitions). For more \
-information, read the 'setup' help file from the next menu." 10 64
+    dialog --title "NO LINUX PARTITIONS DETECTED" \
+    --msgbox "There don't seem to be any partitions on this machine of type \
+    Linux or Bcachefs. You'll need to make at least one of these to install Linux. \
+    To do this, you'll need to leave 'setup' and make the partitions using \
+    'cfdisk' (MBR partitions) or 'cgdisk' (GPT partitions). For more \
+    information, read the 'setup' help file from the next menu." 10 64
 fi
 if [ -d /sys/firmware/efi ]; then
   # First, let's make sure that efivarfs is active:
